### PR TITLE
refactor: tools

### LINKED ISSUE #1656: Pressing 'a' on a FID
Currently when we press the shortcut 'a' we select the phase correction even if this tool is not available.

For instance in the case of a FID you should never be able to select the phase correction tool.

I would prefer in the case of a 'fid' that 'a' selects the 'apodization' tool.

### PR DESCRIPTION
**Pull request by hamed-musallam**
Changes:

1. remove the "tool" suffix from the tool names and reflect the changes and change another

Old tools options 
```
  zoomTool: boolean;
  zoomOutTool: boolean;
  import: boolean;
  exportAs: boolean;
  spectraStackAlignments: boolean;
  spectraCenterAlignments: boolean;
  realImaginary: boolean;
  peakTool: boolean;
  integralTool: boolean;
  zonePickingTool: boolean;
  slicingTool: boolean;
  rangePickingTool: boolean;
  zeroFillingTool: boolean;
  apodizationTool: boolean;
  phaseCorrectionTool: boolean;
  baselineCorrectionTool: boolean;
  FFTTool: boolean;
  multipleSpectraAnalysisTool: boolean;
  exclusionZonesTool: boolean;
```

  New tools options

```
  zoom: boolean;
  zoomOut: boolean;
  import: boolean;
  exportAs: boolean;
  spectraStackAlignments: boolean;
  spectraCenterAlignments: boolean;
  realImaginary: boolean;
  peakPicking: boolean;
  integral: boolean;
  zonePicking: boolean;
  slicing: boolean;
  rangePicking: boolean;
  zeroFilling: boolean;
  apodization: boolean;
  phaseCorrection: boolean;
  baselineCorrection: boolean;
  fastFourierTransform: boolean;
  multipleSpectraAnalysis: boolean;
  exclusionZones: boolean;
```

3. improve the useCheckToolVisibility hook and centralize the tool checking based on Display mode and if the spectrum is selected or not and if it is FID or FT spectrum
4. use the new useCheckToolVisibility functionality and reflect the usage in the other components like KeysListenerTracker and basic and function toolbar

close #1656

### CODE REVIEW

**Review by lpatiny — APPROVED** [review]
LGTM but it could still change in the future I guess

File: src/component/elements/formik/FormikOnChange.tsx [hunk 1]
```diff
@@ -8,11 +8,14 @@ interface FormikOnChangeProps {
...
   const { onChange = () => null, enableValidation = true } = props;
+  const oldValuesRef = useRef({});
   const { values, errors, submitForm, setTouched } = useFormikContext();
-
   useEffect(() => {
     if (enableValidation) {
-      if (Object.keys(errors).length === 0) {
+      if (
+        Object.keys(errors).length === 0 &&
+        JSON.stringify(oldValuesRef.current) !== JSON.stringify(values)
```
Comment (lpatiny) [inline src/component/elements/formik/FormikOnChange.tsx:17]: Why not using directly an empty object rather than a complex useRef ?

Comment (hamed-musallam) [reply src/component/elements/formik/FormikOnChange.tsx:17]: Because i need to see the previous value and compare the new with the old one

Comment (hamed-musallam) [reply src/component/elements/formik/FormikOnChange.tsx:17]: the aim of this component is to listen to the formik changes and trigger onChange callback by this i don not need to listen for each input

Comment (lpatiny) [reply src/component/elements/formik/FormikOnChange.tsx:17]: And where to you change the value of oldValuesRef ? There is only 2 instances of this variable in the whole project

image

**Pull request merged**